### ISSUE TITLE
Check if .NET SDK is installed on plugin add

### ISSUE THREAD

**Issue by AgentEnder** [AgentEnder/nx-dotnet#3]
1.  During installation check if .NET Sdk is installed
    1.  Check against package.json sdk version
    2. Warn if mismatch
2. Supply link to current version

**Comment by Jordan-Hall** [comment]
download link to the binaries dotnet.microsoft.com/download/dotnet/thank-you/sdk-[5.0.200]-[macos/windows/linux]-[cpu]-binaries my suggested we download the binaries and place them into .bin folder. The alternative is to create another package which we publish fist and then use typescript bash to wrap it.

**Comment by AgentEnder** [comment]
My only concern with this approach would be keeping that version number up to date. It would not be too big of a burden to update the plugin to support latest version every now and again, but is something to consider. We should definitely use global version rather than install sdk locally if available and version matches though, .net sdk doesn't need to be cloned all over a user's pc.

**Comment by Jordan-Hall** [comment]
Just a note in that one. Microsoft has a version json of the latest version and download link. Maybe that's a better route.

**Comment by AgentEnder** [comment]
Ah, that could definitely work well. Do you have a link to that json file?

**Comment by Jordan-Hall** [comment]
https://raw.githubusercontent.com/dotnet/core/main/release-notes/releases-index.json
https://dotnetcli.blob.core.windows.net/dotnet/release-metadata/5.0/releases.json

We could have like version 5 then depending on if its ^ or ~ depends on what we actually provide. Another way is to then make a npm package compile on each update so a team can fix a version too

**Comment by AgentEnder** [comment]
Thats very interesting, we will need to decide for versioning whether to version the plugin based off of Nx's major version or .NET's major version. It would certainly make it easier for users to update their package.json if for example we launched at 11.0.0 since Nx is at  11.x.y, but that might not be a huge help.

I wish there was a spot in that json that stated if it was a preview, release candidate or stable.

**Comment by ntziolis** [comment]
In my experience when using nx-plugins, it can be helpful when they are versioned in line with nx as this eliminates the need to figure out if a plugin version is compatible with the version of nx. That said having a table on the repo page with nx version to plugin version range easily fixes that + there is also the warning from npm for incorrect peer deps.

Also removing the control of the major version from the naming scheme does limit the plugin lib team to be able to publish breaking changes without an nx major version change.

Tying the plugin versioning to the  major version of the downstream tech would be very unusual looking at the other nx plugins. Ans also kind of feels backwards. It might also be that a one version of the plugin does support multiple versioning of the downstream tech (there are some nx plugins that do that today).

So in my opinion you should:
- Either go for completely entirely separate sem versioning
- Or tie it to major version of nx

**Comment by AgentEnder** [comment]
Yeah, the versioning will be standalone since we have semantic-release setup now. Currently we are not doing anything for this issue, but at minimum doing something like:

`dotnet --version` and giving an error if dotnet is not found would be enough to close this out imho.
